sdk/servicebus/azure-servicebus/api_review.py

```diff
@@ -0,0 +1,341 @@
+    def from_topic(
+        cls,
+        fully_qualified_namespace : str,
+        topic_name : str,
+        credential : TokenCredential,
+        **kwargs
+    ) -> SeviceBusSenderClient:
+    @classmethod
+    def from_connection_string(
+        cls,
+        conn_str : str,
+        entity_name : str = None,
+        **kwargs
+    ) -> SeviceBusSenderClient:
+
+    def __enter__(self):
+    def __exit__(self):
+
+    def close(self) -> None:
+
+    def get_properties(self) -> Dict[str, Any]:
+
+    def list_session_ids(self) -> List[str]:
+
+    def create_batch(
+        self,
+        session_id : str = None,
+        max_size_in_bytes : int = None
+    ) -> BatchMessage:
+
+    def send(
+        self,
+        message : Union[Message, BatchMessage],
+        session_id : str = None,
+        message_timeout : float = None
+    ) -> None:
+    def schedule(
+        self,
+        message : Union[Message, BatchMessage],
+        schedule_time : datetime,
+        session_id : str = None
+    ) -> List[int]:
+
+    def cancel_scheduled_messages(self, sequence_number : Union[int, List[int]]) -> None:
+
+
+class ServiceBusReceiverClient:
+    def __init__(
+        self,
+        fully_qualified_namespace : str,
+        entity_name : str,
+        credential: TokenCredential,
+        subscription_name : str = None,
+        session_id : str = None,
+        mode : ReceiveSettleMode = ReceiveSettleMode.PeekLock
+        logging_enable: bool = False,
+        http_proxy: dict = None,
+        transport_type: TransportType = None,
+        retry_total : int = 3,
+        retry_backoff_factor: float = 30,
+        retry_backoff_maximum: int = 120  # cur_retry_backoff = retry_backoff_factor * (2^cur_retry_time)
+        connection: Connection = None
+    ) -> None:
+    @classmethod
+    def from_queue(
+        cls,
+        fully_qualified_namespace : str,
+        queue_name : str,
+        credential : TokenCredential,
+        session_id: str = None,
+        mode : ReceiveSettleMode = ReceiveSettleMode.PeekLock
+        **kwargs
+    ) -> ServiceBusReceiverClient:
+    @classmethod
+    def from_topic_subscription(
+        cls,
+        fully_qualified_namespace : str,
+        topic_name : str,
+        subscription_name : str,
+        credential: TokenCredential,
+        session_id: str = None,
+        mode : ReceiveSettleMode = ReceiveSettleMode.PeekLock
+        **kwargs
+    ) -> ServiceBusReceiverClient:
+    @classmethod
+    def from_connection_string(
+        cls,
+        conn_str : str,
+        entity_name : str = None,
+        subscription_name : str = None,
+        session_id: str = None,
+        mode : ReceiveSettleMode = ReceiveSettleMode.PeekLock
+        **kwargs
+    )-> ServiceBusReceiverClient:
+
+    def __enter__(self):
+    def __exit__(self):
+
+    def close(self) -> None:
+
+    def get_properties(self) -> Dict[str, any]:
+
+    def __iter__(self):
+    def __next__(self):
+    def next(self):
```

Comment:
option: add a method like `get_receiver_iterator(auto_complete, auto_lock_renew)` which returns the iterator.

Reply:
Closing this as no current reason to change our iterator pattern.